packages/backend/src/models/SavedChartModel.ts

```diff
@@ -469,6 +554,16 @@ export class SavedChartModel {
                 )
                 .orderBy('saved_queries_versions.created_at', 'desc')
                 .limit(1);
+
+            if (versionUuid) {
+                chartQuery.where(
+                    `${SavedChartVersionsTableName}.saved_queries_version_uuid`,
+                    versionUuid,
+                );
```

Comment:
This method was already getting the chart info based on the latest version. So we just need to filter by a version uuid.